crates/biome_analyze/CONTRIBUTING.md

```diff
@@ -103,54 +105,154 @@ New rules **must** be placed inside the `nursery` group. This group is meant as
 
-### Creating and implementing the rule
+### Creating and Implementing the Rule
 
 Let's say we want to create a new **lint** rule called `useMyRuleName`, follow these steps:
 
-1. Run the command
+1. **Generate the code for your rule** by running this command
+   _(Hint: Replace `useMyRuleName` with your custom name as recommended by the [naming convention](#guideline-naming-convention-for-rules))_:
 
    ```shell
+   # Example: Create a new JS lint rule
    just new-js-lintrule useMyRuleName
+
+   # Or, to create a new lint/assist rule for JSON/CSS/GraphQL/..:
+   # $ just new-css-lintrule useMyRuleName
+   # $ just new-graphql-lintrule useMyRuleName
+   # $ just new-js-assistrule useMyRuleName
+   # $ just new-json-assistrule useMyRuleName
    ```
-   The script will generate a bunch of files for the _JavaScript_ language, inside the `biome_js_analyze` crate.
-   Among the other files, you'll find a file called `use_my_new_rule_name.rs` inside the `biome_js_analyze/lib/src/lint/nursery` folder. You'll implement your rule in this file.
+   The script `just new-js-lintrule` script will generate a bunch of files for the _JavaScript_ language inside the `biome_js_analyze` crate.
+   Among the other files, you'll find a file called `use_my_rule_name.rs` inside the `biome_js_analyze/lib/src/lint/nursery` folder. You'll implement your rule in this file.
 
-   If you want to create a _CSS_ lint rule, run this script. It will generate a new lint rule in `biome_css_analyze`
+2. Let's have a look at the generated code in  `use_my_rule_name.rs`:
 
-   ```shell
-   just new-css-lintrule useMyRuleName
+
+   ```rust
+   ...
+   impl Rule for UseMyRuleName {
+       type Query = Ast<JsIdentifierBinding>;
+       type State = ();
+       type Signals = Option<Self::State>;
+       type Options = ();
+
+       fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+           ...
+       }
+
+       fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+           ...
+       }
+   }
    ```
 
-1. The `Ast` query type allows you to query the CST of a program.
-1. The `State` type doesn't have to be used, so it can be considered optional. However, it has to be defined as `type State = ()`.
-1. Implement the `run` function:
+   We can observe a few details from this snippet:
+
+   - The **`Options`** type can be used to define [additional options](#rule-options) for your rule:
+
+     ```rust
+     type Options = ();
+     ```
+
+     Use `()` if your rule does not have additional options.
+
+   - The **`Query`** type defines the entities for which your your rule's `UseMyRuleName::run` function will be invoked:
+     ```rust
+     type Query = Ast<JsIdentifierBinding>;
+     ```
+
+     _→ The `Ast<>` query type for example allows you to query the AST/CST of a program._
```

Comment:
```suggestion
     _→ The `Ast<>` query type, for example, allows you to query the node/token of a program._
```

Reply:
I'd say that this would not be entirely correct, because we are **querying** the AST **for** a specific node/type. So maybe this?
```suggestion
     _→ The `Ast<>` query type, for example, allows you to query the AST/CST of a program for nodes of a specific type._
```